src/workerd/util/duration-alert.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "kj/string.h"
+#include "kj/time.h"
+#include <kj/debug.h>
+
+class durationAlert {
+// This is a utility class for instantiating a timer, which will log if it's destructed after a specified time
+// This works by relying on RAII (Scope-Bound Resource Management) to check how much time has elapsed
+// when the object is destructed. This ensures that the time is checked when the timer object goes out of scope
+// thereby timing everything after the initialization of the timer, until the end of the scope where it was
+// instantiated.
+public:
+  durationAlert(const kj::MonotonicClock& clock, kj::Duration logWarningTime, kj::StringPtr logMessage)
+    : logWarningTime(logWarningTime),
+      logMessage(logMessage),
+      start(clock.now()),
+      clock(clock) {}
+
+  ~durationAlert() {
```

Comment:
Per the style guide, explicit destructors should typically be declared `noexcept(false)`.

https://github.com/capnproto/capnproto/blob/master/style-guide.md#exceptions-usage

Reply:
Thanks! Marked it as noexcept(false), but I didn't include the UnwindDetector. I think that's fine?

Reply:
Yeah, I think that's fine.  It seems like UnwindDetector is mostly used in places that either explicitly throw exceptions or could be reasonably anticipated to throw exceptions.